codec/src/test/java/io/netty/handler/codec/compression/ZstdEncoderTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.compression;
+
+import com.github.luben.zstd.ZstdInputStream;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.embedded.EmbeddedChannel;
+import org.junit.jupiter.api.BeforeEach;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.InputStream;
+
+
+import static org.mockito.Mockito.when;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ZstdEncoderTest extends AbstractEncoderTest {
+
+    @Mock
+    private ChannelHandlerContext ctx;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        when(ctx.alloc()).thenReturn(ByteBufAllocator.DEFAULT);
+    }
+
+    @Override
+    public EmbeddedChannel createChannel() {
+        return new EmbeddedChannel(new ZstdEncoder());
+    }
+
+    @Override
+    public void testCompressionOfBatchedFlow(final ByteBuf data) throws Exception {
+        final int dataLength = data.readableBytes();
+        int written = 0;
+
+        ByteBuf in = data.retainedSlice(written, 65535);
```

Comment:
What happens if data.capacity() < 65535 ?

Reply:
`ZstdEncoder` will work normally.I changed `testCompressionOfBatchedFlow` to `testCompressionOfLargeBatchedFlow`, keeping the original implementation, thanks